pkg/policy/api/rule_validation_test.go

```diff
@@ -1174,69 +1178,90 @@ func BenchmarkCIDRSanitize(b *testing.B) {
 }
 
 func TestSanitizeDefaultDeny(t *testing.T) {
+	trueValue := true
+	falseValue := false
 	for _, tc := range []struct {
-		before      Rule
-		wantIngress bool
-		wantEgress  bool
+		before    Rule
+		wantError bool
 	}{
 		{
-			before: Rule{},
+			before:    Rule{},
+			wantError: true,
+		},
+		{
+			before: Rule{
+				Ingress:           []IngressRule{{}},
+				EnableDefaultDeny: DefaultDenyConfig{Egress: &trueValue},
+			},
+			wantError: true,
+		},
+		{
+			before: Rule{
+				Ingress:           []IngressRule{{}},
+				EnableDefaultDeny: DefaultDenyConfig{Egress: &falseValue},
+			},
+			wantError: true,
+		},
+		{
+			before: Rule{
+				Egress:            []EgressRule{{}},
+				EnableDefaultDeny: DefaultDenyConfig{Ingress: &trueValue},
+			},
+			wantError: true,
+		},
+		{
+			before: Rule{
+				Egress:            []EgressRule{{}},
+				EnableDefaultDeny: DefaultDenyConfig{Ingress: &falseValue},
+			},
+			wantError: true,
 		},
 		{
 			before: Rule{
 				Ingress: []IngressRule{{}},
 			},
-			wantIngress: true,
 		},
 		{
 			before: Rule{
 				IngressDeny: []IngressDenyRule{{}},
 			},
-			wantIngress: true,
 		},
 		{
 			before: Rule{
 				Ingress:     []IngressRule{{}},
 				IngressDeny: []IngressDenyRule{{}},
 			},
-			wantIngress: true,
 		},
 		{
 			before: Rule{
 				Egress:     []EgressRule{{}},
 				EgressDeny: []EgressDenyRule{{}},
 			},
-			wantEgress: true,
 		}, {
 			before: Rule{
 				EgressDeny: []EgressDenyRule{{}},
 			},
-			wantEgress: true,
 		},
 		{
 			before: Rule{
 				Egress: []EgressRule{{}},
 			},
-			wantEgress: true,
 		},
 		{
 			before: Rule{
 				Egress:  []EgressRule{{}},
 				Ingress: []IngressRule{{}},
 			},
-			wantEgress:  true,
-			wantIngress: true,
 		},
 	} {
 		b := tc.before
 		b.EndpointSelector = EndpointSelector{LabelSelector: &slim_metav1.LabelSelector{}}
 
 		err := b.Sanitize()
-		assert.NoError(t, err)
-		assert.NotNil(t, b.EnableDefaultDeny.Egress)
-		assert.NotNil(t, b.EnableDefaultDeny.Ingress)
-
-		assert.Equal(t, tc.wantEgress, *b.EnableDefaultDeny.Egress, "Rule.EnableDefaultDeny.Egress should match")
-		assert.Equal(t, tc.wantIngress, *b.EnableDefaultDeny.Ingress, "Rule.EnableDefaultDeny.Ingress should match")
```

Comment:
It looks like this test takes the public version of a rule without `EnableDefaultDeny` and then sanitizes the rule for internal use, which populates the `EnableDefaultDeny.Ingress` if there is an ingress rule (and equivalent for egress).

However now with the latest changes, the "default deny" property is no longer tested but it's being replaced with a test for the number of rules.

I think that ideally this test validates both things - ensure that the default deny behavior is correct post-sanitization even if the user did not explicitly specify it, _and_ validate the error checking for the new cases you've introduced.

Reply:
done.